### PR TITLE
TASK-39: Setup training pipeline and add docs

### LINKED ISSUE #143: TASK-39: Setup Machine Learning Training
**Task Description**

As a developer, I want to setup training a model so that it can be used in the application.

**Subtasks**

- [ ] 39.1 Gather Data
- [ ] 39.2 Setup training pipeline: From data and training, get a model that can recognize objects and run in the browser
- [ ] 39.3 Setup Validation of the model

**Acceptance Criteria**

- [ ] Get a model that can be used in the app
- [ ] Documentation present where necessary

### PR DESCRIPTION
**Pull request by cristian-aldea**
closes #143 

This PR aims to add the infrastructure and documentation needed to train an object detection model for the Mull project.

* Adds folders needed for training (currently empty, populated during data gathering and training)
* Adds documentation for the process of training, exporting and converting the model, to get a model that can run in the browser.

To review this PR, just read over the `README.md` and make sure the instructions are understandable (even if you're not familiar with these technologies. If something is confusing, please reach out to me.)

If you are interested in learning more about this technology, I can guide you through the process I took to train a model, but obviously there is no need for everyone to do that, as it takes a lot of time. 

I have made the models that i trained available at [TheGreatMarkus/mull-model](https://github.com/TheGreatMarkus/mull-model) if you're interested.

Also, we have [a OneDrive folder](https://liveconcordia-my.sharepoint.com/:f:/g/personal/c_aldea_live_concordia_ca/Evd3mpAYFE1MjbhUcxKisUQBRz2IVUegTL8FdK0yFa3ztg?e=XB4G2V) used as the repository for all of our training data. Password is on Slack

### CODE REVIEW

**Comment by cristian-aldea** [comment]
Sorry about the notifications, converted this to a draft for now, still testing a few things

**Comment by cristian-aldea** [comment]
Opened back this PR. I tried my best to get the documentation to be easy to understand for someone who is somewhat familiar with machine learning. It should also still be comprehensible for someone with no experience in machine learning.

My goal is to have this part of the project understood by the team, so please reach out if you have any questions.

**Review by ys-lin — APPROVED** [review]
- [x] Get a model that can be used in the app
- [x] Documentation present where necessary

**Review by ys-lin — APPROVED** [review]
- [x] Get a model that can be used in the app
- [x] Documentation present where necessary
- [x] 39.3 Setup Validation of the model

File: ml/trash-recognition/README.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+### Validate Model
+
+```bash
+# Run this in parallel in another terminal. It will run tests and evaluate the model every few minutes
+# Keep in mind that this process will also take up some GPU memory, so if you're having issues with that you can just run it after training is done
+python model_main_tf2.py --model_dir=models/$MODEL_NAME --pipeline_config_path=models/$MODEL_NAME/pipeline.config --checkpoint_dir=models/$MODEL_NAME
+```
+
+### Export Model as a SavelModel
```
Comment (cindyslittleplanet) [inline ml/trash-recognition/README.md:84]: Smol typo here, should be SavedModel

**Review by cindyslittleplanet — APPROVED** [review]
Very complete and informative README; From definition of each folder to troubleshooting sections. 

- [x] Get a model that can be used in the app
- [x] Documentation present where necessary

**Pull request merged**